lib/utils.pm

```diff
@@ -267,33 +267,26 @@ sub zypper_call {
     my $log              = $args{log};
     my $dumb_term        = $args{dumb_term};
 
-    my $str = hashed_string("ZN$command");
-    my $redirect = is_serial_terminal() ? '' : " > /dev/$serialdev";
-
-    if ($log) {
-        script_run("zypper -n $command | tee /tmp/$log; echo $str-\${PIPESTATUS}-$redirect", 0);
-    }
-    elsif ($dumb_term) {
-        script_run("zypper -n $command | cat; echo $str-\${PIPESTATUS}-$redirect", 0);
-    }
-    else {
-        script_run("zypper -n $command; echo $str-\$?-$redirect", 0);
+    my $printer = $log ? "| tee /tmp/$log" : $dumb_term ? '| cat' : '';
+
+    # Retrying workarounds
+    my $ret;
+    for (1 .. 3) {
+        $ret = script_run("zypper -n $command $printer; ( exit \${PIPESTATUS[0]} )", $timeout);
+        if ($ret == 502) {
+            record_soft_failure "Retrying because of error 502";
```

Comment:
please no soft failure without bug reference. If there is no product bug to solve, the soft fail is worthless